tests/src/main/java/net/neoforged/neoforge/debug/entity/TradeTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.debug.entity;
+
+import java.util.ArrayList;
+import net.minecraft.core.BlockPos;
+import net.minecraft.gametest.framework.GameTest;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.npc.VillagerData;
+import net.minecraft.world.entity.npc.VillagerProfession;
+import net.minecraft.world.entity.npc.VillagerTrades;
+import net.minecraft.world.entity.npc.VillagerType;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.neoforged.bus.api.IEventBus;
+import net.neoforged.fml.common.Mod;
+import net.neoforged.neoforge.common.BasicItemListing;
+import net.neoforged.neoforge.common.NeoForge;
+import net.neoforged.neoforge.event.village.VillagerTradesEvent;
+import net.neoforged.testframework.DynamicTest;
+import net.neoforged.testframework.annotation.ForEachTest;
+import net.neoforged.testframework.annotation.TestHolder;
+import net.neoforged.testframework.gametest.EmptyTemplate;
+
+@ForEachTest(groups = TradeTests.GROUP)
+public class TradeTests {
+    public static final String GROUP = "level.entity.trades";
+    private static final String MOD_ID = "villager_trade_test";
+
+    @Mod(value = MOD_ID)
```

Comment:
Why is this a mod and not a static event listener?

Reply:
Don't all the subscribe annotations require it to be attached to a mod. I cannot find an example of a static event listener in any of the new tests so I will need an example of what you mean if not the annotations

Reply:
Test fixed and made more robust to ensure we don't lost vanilla trades by mistake